ldk/go/file.go

```diff
@@ -0,0 +1,13 @@
+package ldk
+
+import "os"
+
+// File this is a file
+type File interface {
+	Read([]byte) (int, error)
```

Comment:
I've been looking at what is standard in C# and this _should_ be close enough to the `FileStream` class and the `System.IO.Stream` abstract class, though we'll likely have to make a custom interface for this as there is no actual `FileStream` interface (just a class).

Reply:
That's good news!

Reply:
The only other thing that jumps out at me here is that C# has ReadAsync, WriteAsync as well, which seem to fit better with an event loop (and in NodeJS, they will pretty much have to be async calls, due to the single thread). I don't know my GoLang enough to know if that (async v synch) is basically what you get with the above implicitly or not?

Reply:
I think the approach here in Go is closer to ReadAsync, WriteAsync. 

If would consider reading the whole file in a singe operation to the non-async analog in Go.